ambari-server/src/main/java/org/apache/ambari/server/topology/ConfigurableHelper.java

```diff
@@ -0,0 +1,186 @@
+
+        checkArgument(item.size() == 1, "Each config object must have a single property which is the name of the config," +
+          " e.g. \"cluster-env\" : {...}");
+
+        Map.Entry<String, ?> configEntry = item.entrySet().iterator().next();
+        String configName = item.keySet().iterator().next();
+
+        checkArgument(configEntry.getValue() instanceof Map,
+          "The value for %s must be a JSON object (found: %s)", configName, getClassName(configEntry.getValue()));
+
+        Map<String, Object> configData = (Map<String, Object>) configEntry.getValue();
+
+        Set<String> extraKeys = Sets.difference(configData.keySet(), PERMITTED_CONFIG_FIELDS);
+        checkArgument(extraKeys.isEmpty(), "Invalid fields in %s configuration: %s", configName, extraKeys);
+
+        if (configData.containsKey(PROPERTIES_PROPERTY_ID)) {
+          checkMap(PROPERTIES_PROPERTY_ID, configData.get(PROPERTIES_PROPERTY_ID), String.class);
+
+          Map<String, String> properties = (Map<String, String>)configData.get(PROPERTIES_PROPERTY_ID);
+          allProperties.put(configName, properties);
+        }
+        if (configData.containsKey(PROPERTIES_ATTRIBUTES_PROPERTY_ID)) {
+          checkMap(PROPERTIES_ATTRIBUTES_PROPERTY_ID, configData.get(PROPERTIES_ATTRIBUTES_PROPERTY_ID), Map.class);
+          Map<String, Map<String, String>> attributes =
+            (Map<String, Map<String, String>>)configData.get(PROPERTIES_ATTRIBUTES_PROPERTY_ID);
+          attributes.entrySet().forEach( entry -> checkMap(entry.getKey(), entry.getValue(), String.class));
+
+          allAttributes.put(configName, attributes);
+        }
+      });
+
+      configuration = new Configuration(allProperties, allAttributes);
+    }
+    return configuration;
+  }
+
+  /**
+   * Converts {@link Configuration} objects to a collection easily serializable to Json
+   * @param configuration the configuration to convert
+   * @return the resulting collection
+   */
+  public static Collection<Map<String, Map<String, ?>>> convertConfigToMap(Configuration configuration) {
+    Collection<Map<String, Map<String, ?>>> configurations = new ArrayList<>();
+    Set<String> allConfigTypes = Sets.union(configuration.getProperties().keySet(), configuration.getAttributes().keySet());
+    for (String configType: allConfigTypes) {
+      Map<String, Map<String, ? extends Object>> configData = new HashMap<>();
+      if (configuration.getProperties().containsKey(configType)) {
+        configData.put(PROPERTIES_PROPERTY_ID, configuration.getProperties().get(configType));
+      }
+      if (configuration.getAttributes().containsKey(configType)) {
+        configData.put(PROPERTIES_ATTRIBUTES_PROPERTY_ID, configuration.getAttributes().get(configType));
+      }
+      configurations.add(ImmutableMap.of(configType, configData));
+    }
+    return configurations;
+  }
+
+  private static void checkFlattenedConfig(Collection<? extends Map<String, ?>> configs) {
```

Comment:
Parts of this method seems to duplicate similar logic in: https://github.com/apache/ambari/blob/1fae6cf19dfc4f1bfb428eb34480e7382c460db8/ambari-server/src/main/java/org/apache/ambari/server/topology/ConfigurationFactory.java#L53-L70

Can you try to reuse that code (refactor if necessary) to avoid duplication?